### ISSUE TITLE
wgpu-hal Uses Virtual Sizes for Compressed Texture Copies Instead of Physical Sizes

### ISSUE THREAD

**Issue by cwfitzgerald** [gfx-rs/wgpu#1895]
**Is your feature request related to a problem? Please describe.**

Follow up to unify https://github.com/gfx-rs/wgpu/pull/1819/

**Describe the solution you'd like**

We should be using physical sizes for texture copies like WebGPU [specifies](https://gpuweb.github.io/gpuweb/#valid-texture-copy-range):

> copySize.width must be a multiple of blockWidth.
> copySize.height must be a multiple of blockHeight.

For example, when uploading the second mip of a 4x4 compressed texture, the physical size is 4x4 and the virtual is 2x2. We pass 2x2 to wgpu-hal currently, but we should stay consistent with WebGPU. This will make the vulkan backend's job a bit harder, but the dx12 backend easier.

**Describe alternatives you've considered**

Not doing this. Imo this is a non-answer as we need wgpu-hal to stay fairly consistent with WebGPU, just unsafe.

**Comment by kvark** [comment]
This is a bit of a mess - https://github.com/gpuweb/gpuweb/issues/363
TL;DR: switching wgpu-hal to physical sizes would not save any work for wgpu-core, but will require the following backends to do more work:
  - Vulkan
  - OpenGL
  - Metal, but only for buffer-texture copies
